sim/src/image.rs

```diff
@@ -1522,14 +1522,19 @@ fn install_image(flash: &mut SimMultiFlash, slot: &SlotInfo, len: ImageSize,
                 // This computation is incorrect, and we need to figure out the correct size.
                 // c::boot_status_sz(dev.align() as u32) as usize
                 16 + 4 * dev.align()
-            } else {
+            } else if Caps::SwapUsingMove.present() {
                 let sector_size = dev.sector_iter().next().unwrap().size as u32;
                 align_up(c::boot_trailer_sz(dev.align() as u32), sector_size) as usize
+            } else if Caps::SwapUsingScratch.present() {
+                c::boot_trailer_sz(dev.align() as u32) as usize
```

Comment:
I wonder why introduction of this line imposes badly on so many tests.

Reply:
The move swap will require an additional sector (beyond the trailer) in slot 0, as it needs to slide slot 0 down 1 sector. The original computation in the code wasn't correct.

Reply:
^^ That patch should affect only swap-using-scratch builds. I need to check whether SwapUsinMove is taken into account properly.

Reply:
I found out that this code path changes behavior only for the swap using scratch. Rest goes with ImageSize::Given(size).